MdePkg/Test/Mock/Include/GoogleTest/Protocol/MockPciRootBridgeIoProtocol.h

```diff
@@ -0,0 +1,71 @@
+/** @file
+  This file declares a mock of PCI Root Bridge IO Protocol.
+
+  Copyright (c) Microsoft Corporation.
+  SPDX-License-Identifier: BSD-2-Clause-Patent
+**/
+
+#ifndef MOCK_PCIROOTBRIDGEIOPROTOCOL_H
+#define MOCK_PCIROOTBRIDGEIOPROTOCOL_H
+
+#include <Library/GoogleTestLib.h>
+#include <Library/FunctionMockLib.h>
+
+extern "C" {
+  #include <Uefi.h>
+  #include <Protocol/PciRootBridgeIo.h>
+}
+
+//
+// Declarations to handle usage of the Pci Root Bridge Io Protocol by creating mock
+//
+struct MockPciRootBridgeIoConfigAccess {
+  MOCK_INTERFACE_DECLARATION (MockPciRootBridgeIoConfigAccess);
+
+  MOCK_FUNCTION_DECLARATION (
+    EFI_STATUS,
+    MockPciRootBridgeIoRead,
+    (IN EFI_PCI_ROOT_BRIDGE_IO_PROTOCOL              *This,
+     IN     EFI_PCI_ROOT_BRIDGE_IO_PROTOCOL_WIDTH    Width,
+     IN     UINT64                                   Address,
+     IN     UINTN                                    Count,
+     IN OUT VOID                                     *Buffer)
+    );
+  MOCK_FUNCTION_DECLARATION (
+    EFI_STATUS,
+    MockPciRootBridgeIoWrite,
+    (IN EFI_PCI_ROOT_BRIDGE_IO_PROTOCOL              *This,
+     IN     EFI_PCI_ROOT_BRIDGE_IO_PROTOCOL_WIDTH    Width,
+     IN     UINT64                                   Address,
+     IN     UINTN                                    Count,
+     IN OUT VOID                                     *Buffer)
+    );
+};
+
+MOCK_INTERFACE_DEFINITION (MockPciRootBridgeIoConfigAccess);
+MOCK_FUNCTION_DEFINITION (MockPciRootBridgeIoConfigAccess, MockPciRootBridgeIoRead, 5, EFIAPI);
+MOCK_FUNCTION_DEFINITION (MockPciRootBridgeIoConfigAccess, MockPciRootBridgeIoWrite, 5, EFIAPI);
+
+EFI_PCI_ROOT_BRIDGE_IO_PROTOCOL  PCI_ROOT_BRIDGE_IO_PROTOCOL_MOCK = {

```

Comment:
Can you please fully mock out the rest of the protocol functions? If all the functions are fully fleshed out, then it will save the churn of someone else needing to come back and retouch this file later.

Reply:
Completed.